app/src/main/java/com/infomaniak/mail/ui/newMessage/AiViewModel.kt

```diff
@@ -91,13 +95,24 @@ class AiViewModel @Inject constructor(
                     history += AssistantMessage(aiResult.content)
                     SUCCESS
                 } ?: MISSING_CONTENT
-                error?.code == MAX_SYNTAX_TOKENS_REACHED -> PROMPT_TOO_LONG
+                error?.code == MAX_SYNTAX_TOKENS_REACHED -> {
+                    if (shouldTriggerContextTooLongInstead) {
+                        discardPreviousMailContext()
+                        CONTEXT_TOO_LONG
+                    } else {
+                        PROMPT_TOO_LONG
+                    }
+                }
                 error?.code == TOO_MANY_REQUESTS -> RATE_LIMIT_EXCEEDED
                 else -> ERROR
             }
         )
     }
 
+    private fun discardPreviousMailContext() {
+        previousMessageBodyPlainText = null
+    }
```

Comment:
This seems overkill to have a wrapping function here just for a bit of clarification
A small commentary would do the job